dali/python/nvidia/dali/external_source.py

```diff
@@ -303,15 +310,14 @@ def __call__(self, *, source = None, cycle = None, name = None, layout = None, c
 def _is_external_source_with_callback(op_instance):
     return isinstance(op_instance._op, ExternalSource) and op_instance._callback is not None
 
-def external_source(source = None, num_outputs = None, *, cycle = None, name = None, device = "cpu", layout = None, cuda_stream = None):
+def external_source(source = None, num_outputs = None, *, cycle = None, name = None, device = "cpu", layout = None, cuda_stream = None, no_copy = False):
```

Comment:
> as we would need to keep the python reference to the data.

What if user will keep providing the same buffer but just modifies the content of it. I think there are some numpy in-place operations. Do we have any other option?

Reply:
I think that now we have no restrictions on what a callback / iterator / generator can do - it can be a partial function, with state attached to some object.

Reply:
In this case let us leave it up to the user to do the right thing. If he cannot preserve the memory passed then it is his fault.